### PR TITLE
GEODE-7995: Creates concurrencyTest facet

### PR DESCRIPTION
**Pull request by pivotal-jbarrett**
* Moves all concurrency style test under its own facet of testing.
* Move concurrency test framework sources to geode-concurrency-test.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by upthewaterspout** [comment]
Can we stage this is a way that doesn't break people's PRs or lose coverage for a while? Maybe we can merge the gradle changes without moving the tests, then add the pipeline support, and then move the tests.

**Comment by pivotal-jbarrett** [comment]
> Can we stage this is a way that doesn't break people's PRs or lose coverage for a while? Maybe we can merge the gradle changes without moving the tests, then add the pipeline support, and then move the tests.

We could but I think the number of tests and the time between this commit and the pipeline tie in will be very short. I think the risk is very very small.

**Comment by pivotal-jbarrett** [comment]
> I think it would be better to split things up and move the tests after the CI changes are done - doesn't seem too hard to do. The CI changes may take a little time - we have to enhance the stress new test job as well.

With the help of @rhoughton-pivot I have pushed a commit to the CI that we are running through a test pipeline. Will wait for results of that.

File: ci/pipelines/shared/jinja.variables.yml [hunk 1]
```diff
@@ -135,6 +135,18 @@ tests:
   PLATFORM: linux
   RAM: '210'
   name: Upgrade
+- ARTIFACT_SLUG: concurrencytestfiles
+  CALL_STACK_TIMEOUT: '1200'
+  CPUS: '96'
+  DUNIT_PARALLEL_FORKS: '0'
+  EXECUTE_TEST_TIMEOUT: 30m
```
Comment (upthewaterspout) [inline ci/pipelines/shared/jinja.variables.yml:142]: Since we turned off parallel everything, is 30 minutes too short for these tests?

**Comment by pivotal-jbarrett** [comment]
Pipeline changes worked on test pipeline. 
https://concourse.apachegeode-ci.info/teams/main/pipelines/pivotal-jbarrett-feature-geode-7995-c-main/jobs/ConcurrencyTestOpenJDK8/builds/1

File: ci/pipelines/shared/jinja.variables.yml (hunk 1 above)
Comment (pivotal-jbarrett) [reply ci/pipelines/shared/jinja.variables.yml:142]: They finish in under 10 minutes.

**Comment by pivotal-jbarrett** [comment]
> I think this is still missing some changes to include this new test target in stress new test.
> 
> Side note - you made a lot of minor cleanup changes with this PR (maybe with some intellij inspection?). It would be nicer to split those out into a separate PR - especially since your changes are not going to get tested because they are running as part of this new gradle target that CI is not going to run on this PR.

The changes were required for them to compile. That project enforces warnings as errors. The test pipeline successfully built and ran those cleaned up tests.

**Comment by rhoughton-pivot** [comment]
The test jobs are here:

https://concourse.apachegeode-ci.info/teams/main/pipelines/pivotal-jbarrett-feature-geode-7995-c-main/jobs/ConcurrencyTestOpenJDK8/builds/1

https://concourse.apachegeode-ci.info/teams/main/pipelines/pivotal-jbarrett-feature-geode-7995-c-main/jobs/ConcurrencyTestOpenJDK11/builds/1

File: geode-membership/build.gradle [hunk 2]
```diff
@@ -66,7 +65,9 @@ dependencies {
     testImplementation('com.tngtech.archunit:archunit-junit4')
 
     integrationTestImplementation('com.tngtech.archunit:archunit-junit4')
-  
+
+    concurrencyTestImplementation sourceSets.integrationTest.output
```
Comment (rhoughton-pivot) [inline geode-membership/build.gradle:69]: This doesn't feel great. What is the dependency here?

Comment (pivotal-jbarrett) [reply geode-membership/build.gradle:69]: Some "framework" type classes shared between the two. The "correct" approach is to create a shared sub module but it's one class. We could iterate on it. The goal wasn't to solve all the worlds problems today.

**Comment by kirklund** [comment]
> What do we mean by facet in this context?

I think it means (1) src set, (2) gradle target to execute tests in the src set, (3) pipeline job in CI and PR.

**Comment by pivotal-jbarrett** [comment]
Very specifically we use a plugin called facet that creates all these things for us.
https://plugins.gradle.org/plugin/nebula.facet

**Comment by rhoughton-pivot** [comment]
@pivotal-jbarrett Want to pair on reviving this PR?

**Comment by onichols-pivotal** [comment]
this PR appears to be abandoned, can it be closed?

**Comment by onichols-pivotal** [comment]
closed due to lack of activity

**Pull request closed without merging**